### PR TITLE
Update to latest m3db version and dependencies

### PR DESCRIPTION
**Pull request by justinjc**

### CODE REVIEW

File: services/m3coordinator/handler/namespace/add.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func (h *addHandler) add(r *admin.NamespaceAddRequest) (nsproto.Registry, error)
 	}
 
 	protoRegistry := namespace.ToProto(nsMap)
-	_, err = h.store.CheckAndSet(M3DBNodeNamespacesKey, version+1, protoRegistry)
+	_, err = h.store.CheckAndSet(M3DBNodeNamespacesKey, version, protoRegistry)
```
Comment (justinjc) [inline services/m3coordinator/handler/namespace/add.go]: @arnikola this was actually a bug that you caught in #41 regarding returning -1 with an empty Metadata slice.

File: services/m3coordinator/handler/namespace/common.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func Metadata(store kv.Store) ([]namespace.Metadata, int, error) {
 		// set at all is semantically the same as having an empty slice of
 		// metadata and is not a real error state.
 		if err == kv.ErrNotFound {
-			return []namespace.Metadata{}, -1, nil
+			return []namespace.Metadata{}, 0, nil
```
Comment (justinjc) [inline services/m3coordinator/handler/namespace/common.go]: @arnikola good catch 👍

**Comment by justinjc** [comment]
Please take another look at this one after [this commit](https://github.com/m3db/m3coordinator/pull/51/commits/8d655b9d700d6d920d89a2bff39dcb94246b5321) to resolve all the metalint complaints. I made best judgement fixes on a lot of these.

File: storage/fanout/storage.go [hunk 3]
```diff
@@ -118,7 +119,7 @@ func (s *fanoutStorage) Close() error {
 	for idx, store := range s.stores {
 		// Keep going on error to close all storages
 		if err := store.Close(); err != nil {
-			logging.WithContext(nil).Error("unable to close storage", zap.Int("store", int(store.Type())), zap.Int("index", idx))
+			logging.WithContext(context.TODO()).Error("unable to close storage", zap.Int("store", int(store.Type())), zap.Int("index", idx))
```
Comment (nikunjgit) [inline storage/fanout/storage.go]: context.Background() ?

File: plan/physical.go [hunk 4]
```diff
@@ -41,7 +40,7 @@ type ResultOp struct {
 
 // NewPhysicalPlan is used to generate a physical plan. Its responsibilities include creating consolidation nodes, result nodes,
 // pushing down predicates, changing the ordering for nodes
-func NewPhysicalPlan(lp LogicalPlan, storage storage.Storage) (PhysicalPlan, error) {
+func NewPhysicalPlan(lp LogicalPlan) (PhysicalPlan, error) {
```
Comment (nikunjgit) [inline plan/physical.go]: I know we need storage in a future diff, any change you can just skip lint with unparam ?

**Pull request merged**